### ISSUE TITLE
onClick is not fired on layer when using layer from existing stack with mapbox.

### ISSUE THREAD

**Issue by nagermen** [visgl/deck.gl#3727]
I am facing an issue when using  layer from an existing Deck's layer stack with mapbox as described [here](https://deck.gl/#/documentation/submodule-api-reference/deckgl-mapbox/mapbox-layer?section=make-a-layer-from-scratch). I am using it within my Angular 8 application here is a [sandbox](https://codesandbox.io/s/angular-kumde?fontsize=14) for it. The problem is: on my IconLayer the handler for onClick event is not fired when layer is clicked, but it is fired on Decks handler. Also there is not that much of an info inside the event data as well. Any help would be highly appreciated cause I am struggling with this one for couple of days already.
Thanks.

#### Repro Steps
1. Open the [sandbox](https://codesandbox.io/s/angular-kumde?fontsize=14) 
2. click on `add data to map` button the icons will appear on map
3. click on any icon and check the console.
You will see that only the handler for Deck is fire while handler for IconLayer is not.

#### Environment (please complete the following information):
 - Angular v8.2.8
 - deck.gl v7.2.6
 - mapboxgl v1.3.1
 - Browser Version: Chrome v77 (happens on Safari as well)

here is a github repo as well. https://github.com/nagermen/deck.gl-mapbox-angular8

**Comment by 1chandu** [comment]
The reason, `onClick` is not getting fired is, layer is not being picked.

**Comment by 1chandu** [comment]
Can you make sure you are using dek.gl 7.2.6, and also try using latest release 7.3.2 ?

From your code pen example, it seem to be using , 7.2.5 and 7.2.6 includes a picking related fix.

<img width="1509" alt="Screen Shot 2019-10-08 at 11 21 03 AM" src="https://user-images.githubusercontent.com/9502731/66422010-c2ee5200-e9bd-11e9-91fe-8f5882245b0a.png">

you can set `deck.log.priority=1` in browser console to log the version.

**Comment by 1chandu** [comment]
I am seeing 7.3.1, can you double check your version, is the sandbox updated?

To debug you can put a breakpoint here, where layer's onClick should be triggered: https://github.com/uber/deck.gl/blob/c185ba2e419bc9347b4ca5df6a6aa571c4c15e8b/modules/core/src/lib/deck.js#L780

**Comment by nagermen** [comment]
In package.json it is hard version for 7.3.2 I don't know why it shows 7.3.1 for you, I will try to figure that out. However this means that the picking related fix is there, but I still see same issue. 
Thanks for a hint, will try to go over it with debugger.

**Comment by 1chandu** [comment]
Once the versioning is fixed we can confirm if there is a picking bug or not, for now I am not able to reproduce this with any standard example, can you give it a try with any standard example in the repo. You can debug from above point and let us know if you need more help.

**Comment by nagermen** [comment]
Sorry for late response, development on mapbox was frozen.
after some debug I noticed that my issue is here 
https://github.com/uber/deck.gl/blob/c185ba2e419bc9347b4ca5df6a6aa571c4c15e8b/modules/core/src/lib/deck.js#L763
after some debugging I noticed that info.layer is `null` here, and because of that layerHandler is `null` over here as well
https://github.com/uber/deck.gl/blob/c185ba2e419bc9347b4ca5df6a6aa571c4c15e8b/modules/core/src/lib/deck.js#L774 so event is not triggered.

will check same on one of the standard examples and create new sandbox for it

**Comment by nagermen** [comment]
Here is the [codepen](https://codepen.io/nagermen/pen/vYYPwYM) with the standard deck.gl ScatterplotLayer example with mapbox.

**Comment by jameswesc** [comment]
Did you find a solution @nagermen ? I'm having the same issue with deck version 8.0.17 with code pretty much the same as the above codepen.

**Comment by nagermen** [comment]
@jameswesc ended up not using layers from the existing stack. Just as shown in the first example here 
https://deck.gl/#/documentation/submodule-api-reference/deckgl-mapbox/mapbox-layer . 
I have checked all the versions of deck.gl from 7.3 till 8 and the issue is still there, please let me know if you find a solution.

**Comment by nagermen** [comment]
@jameswesc  did you maybe found any solutions to this? still facing it on version 8.1.1

**Comment by Pessimistress** [comment]
@nagermen This is because the deck instance does not have the correct `viewState`. 

Technically speaking, deck and mapbox's viewports are two separate systems and have to be synced programmatically. When you use `MapboxLayer` without an existing stack, the layer internally listens to mapbox's `move` event and updates the internal Deck instance. When you use layers from an existing stack, however, this module does not sync the two automatically, because it does not know which one is "driving". The primary use case for an external deck instance is the React module, and the synchronization is handled in the React component.

This is indeed a bug, but cannot be easily fixed without revisiting some of the fundamental design assumptions of the mapbox module. A work around is to manually sync deck's view state to that of the mapbox map:

```js
map.on('move', () => {
  const {lng, lat} = map.getCenter();
  deck.setProps({
    viewState: {
      longitude: lng,
      latitude: lat,
      zoom: map.getZoom(),
      bearing: map.getBearing(),
      pitch: map.getPitch()
    }
  });
  // Prevent deck from redrawing - repaint is driven by mapbox's render loop
  deck.needsRedraw({clearRedrawFlags: true});
});
```

**Comment by nagermen** [comment]
@Pessimistress thanks a lot for your response. I like this workaround solution, will try it.

**Comment by pdfabbro** [comment]
I am having the same issue.  @nagermen did the workaround work for you?  I've tried both the workaround and this without luck https://deck.gl/#/documentation/submodule-api-reference/deckgl-mapbox/mapbox-layer

**Comment by Pessimistress** [comment]
Fixed in v8.6.9.
